util/util.go

```diff
@@ -284,7 +284,11 @@ func EncodeString(value string, algorithm string, salt string) string {
 
 	h.Write([]byte(text))
 
-	return hex.EncodeToString(h.Sum(nil))
+	result := hex.EncodeToString(h.Sum(nil))
+	if len(result) == h.Size()*2 {
```

Comment:
sorry to mislead you but since it is considered as a number this length check is not needed anymore actually.